plugins/node/opentelemetry-instrumentation-awslambda/src/awslambda.ts

```diff
@@ -0,0 +1,178 @@
+        // NB: The patching infrastructure seems to match names backwards, this must be the filename, while
+        // InstrumentationNodeModuleFile must be the module name.
+        filename,
+        ['*'],
+        undefined,
+        undefined,
+        [
+          new InstrumentationNodeModuleFile(
+            module,
+            ['*'],
+            (moduleExports: LambdaModule) => {
+              diag.debug('Applying patch for lambdatest handler');
+              if (isWrapped(moduleExports[functionName])) {
+                this._unwrap(moduleExports, functionName);
+              }
+              this._wrap(moduleExports, functionName, this._getHandler());
+              return moduleExports;
+            },
+            (moduleExports?: LambdaModule) => {
+              if (moduleExports == undefined) return;
+              diag.debug('Removing patch for lambdatest handler');
+              this._unwrap(moduleExports, functionName);
+            }
+          ),
+        ]
+      ),
+    ];
+  }
+
+  private _getHandler() {
+    return (original: Handler) => {
+      return this._getPatchHandler(original);
+    };
+  }
+
+  private _getPatchHandler(original: Handler) {
+    diag.debug('patch handler function');
+    const plugin = this;
+
+    return function patchedHandler(
+      this: {},
+      event: {},
+      context: Context,
+      callback: Callback
+    ) {
+      const name = context.functionName;
+      const span = plugin.tracer.startSpan(name, {
+        kind: SpanKind.SERVER,
+        attributes: {
+          [FaasAttribute.FAAS_EXECUTION]: context.awsRequestId,
+          [FaasAttribute.FAAS_ID]: context.invokedFunctionArn,
+          // TODO(anuraaga): Add cloud.account.id when there is a JS accessor for it.
+        },
+      });
+
+      // Lambda seems to pass a callback even if handler is of Promise form, so we wrap all the time before calling
+      // the handler and see if the result is a Promise or not. In such a case, the callback is usually ignored.
+      const wrappedCallback = plugin._wrapCallback(callback, span);
+      const maybePromise = safeExecuteInTheMiddle(
+        () => original.apply(this, [event, context, wrappedCallback]),
+        error => {
+          if (error != null) {
+            span.setStatus({
+              code: SpanStatusCode.ERROR,
+              message: typeof error === 'string' ? error : error.message,
+            });
+            span.end();
+          }
+        }
+      ) as Promise<{}> | undefined;
+      if (
+        typeof maybePromise !== 'undefined' &&
+        typeof maybePromise.then === 'function'
+      ) {
+        maybePromise.then(
+          () => span.end(),
+          (err: Error | string) => {
+            let errMessage;
+            if (typeof err === 'string') {
+              errMessage = err;
+            } else if (err) {
+              errMessage = err.message;
+            }
+            if (errMessage) {
+              span.setStatus({
```

Comment:
+ `span.recordException`

Reply:
recordException I don't think sets status does it? You may need to do both if that is the behavior you want.